repositories/EventRepository.ts

```diff
@@ -33,16 +33,51 @@ export class EventRepository extends BaseRepository<EventModel> {
   }
 
   public async findByAttendanceCode(attendanceCode: string): Promise<EventModel> {
-    return this.repository.findOne({ attendanceCode });
+    const matchingEvents = await this.repository.find({ attendanceCode });
+
+    // Find all events with the given attendance code
+    const validEvents = matchingEvents.filter((event) => !event.isTooEarlyToAttendEvent() && !event.isTooLateToAttendEvent());
+
+    // If there are eligible events, return the first one
+    if (validEvents.length > 0) {
+      return validEvents[0];
```

Comment:
What defines the first event? Is it sorted by start time?

Reply:
In theory there should never be more than one eligible event; that would mean that there are two events occurring at the same time sharing an attendance code which wouldn't make any sense. This is handled by isAvailableAttendanceCode and tested in the attached postman screenshots. 

However, since the check-in period is defined as 30 minutes before and after an event, two events happening one after another with the same attendance code could cause an issue. 

The solution to this could be to simply disallow this case, which involves changing isAvailableAttendanceCode to check for conflicting events in this "check-in period" rather than the duration of the event itself. Let me know your thoughts on this change!

Reply:
I agree with the "check-in period" approach. However, realistically, we can allow more padding room cases like this, such as not allowing events with the same checkin code 3 days before and after the event. This would avoid the edge case of having a time precisely in the "check-in period"s of two close events with the same checkin code.